ssc/Pos/Ssc/Worker.hs

```diff
@@ -365,14 +380,19 @@ waitUntilSend msgTag epoch slMultiplier = do
             -- TODO [CSL-2173]: Clarify
             leftToPanic "waitUntilSend: " $
             mkLocalSlotIndex $ slMultiplier * fromIntegral slotSecurityParam
-    Timestamp beginning <-
-        getSlotStartEmpatically $
-        SlotId {siEpoch = epoch, siSlot = slot}
-    curTime <- currentTime
-    let minToSend = curTime
-    let maxToSend = beginning + mpcSendInterval
-    when (minToSend < maxToSend) $ do
-        let delta = maxToSend - minToSend
+    timeDiff <- do
+        -- Compute an approximation for the time difference between the current
+        -- slot and the next slot. The result might be inaccurate, i tak soydet.
+        curSlot <- getCurrentSlotInaccurate
+        slotDur <- getNextEpochSlotDuration
+        let
+            targetSlot = SlotId {siEpoch = epoch, siSlot = slot}
+            subzero a b = if a > b then a - b else 0
+            slotDiff = (subzero `on` flattenSlotId) curSlot targetSlot
```

Comment:
Shouldn't `targetSlot` be before `curSlot`?

Reply:
Right, I used `subtract` initially, but then forgot to flip the arguments back.